tests/integration/basic/test_weird_target_dist.py

```diff
@@ -0,0 +1,20 @@
+import unittest
+import pandas as pd
+from lightwood.api.types import ProblemDefinition
+
+
+class TestBasic(unittest.TestCase):
+    def test_0_unkown_cateogires_in_test(self):
+        from lightwood.api.high_level import predictor_from_problem
+
+        # The target will be cateogircal and there will be a bunch of values
+        # in all datasets (train/dev/validation) that were not present in the others
+        df = pd.DataFrame({
+            'target': [1 for _ in range(500)] + [f'{i}cat' for i in range(100)],
+            'y': [i for i in range(600)]
+        })
+        target = 'target'
+
+        predictor = predictor_from_problem(df, ProblemDefinition.from_dict({'target': target, 'time_aim': 200}))
+        predictor.learn(df)
+        predictor.predict(df)
```

Comment:
How about some assert? Maybe assert that the missing targets were encoded as expected?

Reply:
No point, we just want to make sure it doesn't crash when the validation see contains things that aren't in train, that's all. And that is implicitly happening due to the number of distinct targets we generate. Might be worth explaining this better though...

Reply:
Even if it is implicitly happening (which imo it's not very easy to see unless you're familiar with lightwood internals) I think it's better to have 2 different dataframes here to guarantee there's stuff the ICP hasn't seen in training when doing the `predict()` call.

Reply:
Then I'd have to call the training functions one by one instead. But in practice you can test the test works (as I did) by running it on staging, and seeing it fails with the same error listed in the issue